src/events/components/SearchModule/index.tsx

```diff
@@ -1,82 +1,51 @@
-      const newEventTypes = actualValue.map((e) => e.value);
-      if (newEventTypes.length > 0) {
-        setEventTypes(JSON.stringify(newEventTypes));
-      } else {
-        setEventTypes(null);
-      }
-    }
-  };
+  useEffect(() => {
+    dispatch(filterEvents({ query, eventTypes, attendanceFilter, page }));
+  }, [query, String(eventTypes), attendanceFilter, page]);
 
-  const handleToDateClick = (day: DateTime) => {
-    const dateTime = day.set({ hour: 0 });
-    const dateTimeStart = DateTime.fromISO(dateStart || DEFAULT_DATE_START_PARAM);
-
-    if (dateTime > dateTimeStart || dateTime.toISODate() === dateTimeStart.toISODate()) {
-      setDateEnd(dateTime.toISODate());
-    }
-  };
-
-  const handleFromDateClick = (day: DateTime) => {
-    const dateTime = day.set({ hour: 0 });
-    const dateTimeEnd = DateTime.fromISO(dateEnd || DEFAULT_DATE_END_PARAM);
-
-    if (dateTime < dateTimeEnd || dateTime.toISODate() === dateTimeEnd.toISODate()) {
-      setDateStart(dateTime.toISODate());
-    }
-  };
-
-  const onToggleSwitchChange = () =>
-    setNonAttendanceEventsChecked(nonAttendanceEventsChecked === 'false' ? 'true' : 'false');
+  useEffect(() => {
+    dispatch(resetEventPage());
+  }, [query, String(eventTypes), attendanceFilter]);
 
   return (
-    <>
+    <div className={style.container}>
+      <SearchInput defaultValue={query} onChange={(event) => setQuery(event.target.value)} />
       <div className={style.grid}>
-        <SearchInput
-          defaultValue={search || DEFAULT_SEARCH_PARAM}
-          onChange={(event) => setSearch(event.target.value)}
-        />
-        <label className={style.attendanceEvent}>
-          <span>Vis arrangementer uten påmelding</span>
-          <ToggleSwitch onChange={onToggleSwitchChange} checked={!(nonAttendanceEventsChecked === 'false')} />
-        </label>
-        <SelectEventTypes selected={eventTypes ? JSON.parse(eventTypes) : []} onChange={onEventTypesInput} />
-        <div>
-          <div className={style.dateInput}>
-            <DateRangeInput
-              dateEnd={DateTime.fromISO(dateEnd || DEFAULT_DATE_END_PARAM)}
-              dateStart={DateTime.fromISO(dateStart || DEFAULT_DATE_START_PARAM)}
-              handleFromDateClick={handleFromDateClick}
-              handleToDateClick={handleToDateClick}
-            />
-          </div>
-        </div>
+        <SelectEventTypes selected={eventTypes} onChange={setEventTypes} />
+        <select
+          onChange={(event) => setAttendanceFilter(event.target.value as AttendanceFilterType)}
+          value={attendanceFilter}
+        >
+          {Object.entries(ATTENDANCE_FILTERS).map(([value, description]) => (
+            <option key={value} value={value}>
+              {description}
+            </option>
+          ))}
+        </select>
       </div>
-    </>
+    </div>
   );
 };
 
+const selectCurrentPage = () => (state: State) => {
+  return state.events.search.page;
```

Comment:
What is a page?
Is it 1 one page of events from the REST API?

Reply:
It is the pages from the API, yes. We're still using pagination in a way, but we're abstracting it as much as we can away from the user.